### PR TITLE
feat(plugins): Add Sentry Babel plugin for React component annotation

### PR DESCRIPTION
**Pull request by 0Calories**
Replaces the Fullstory Babel annotate plugin with our own version. See https://github.com/getsentry/sentry-javascript-bundler-plugins/pull/468 for more details

This plugin essentially does the same thing, but will annotate the DOM with `data-sentry-component` annotations instead, which is what our SDK looks for when sending component name data to Sentry. This will allow us to start dogfooding new features like span grouping by component names, breadcrumbs, and component names in Replays!

Check out the [NPM Page](https://www.npmjs.com/package/@sentry/component-annotate-plugin) for more details on the plugin

### CODE REVIEW

**Comment by 0Calories** [comment]
Thanks for the heads up @JonasBa , I'll take a look

**Comment by 0Calories** [comment]
I realized that babel plugins must be prefixed with `babel` in their package name, otherwise it can't resolve if you try to add it to your babel config via the package name. I'm working on updating this first before proceeding

**Comment by 0Calories** [comment]
Update: It needs to be prefixed with `babel-plugin`, not `babel` 🤦

File: babel.config.ts [hunk 1]
```diff
@@ -24,7 +25,7 @@ const config: TransformOptions = {
   plugins: [
     '@emotion/babel-plugin',
     '@babel/plugin-transform-runtime',
-    '@babel/plugin-transform-class-properties',
```
Comment (billyvg) [inline babel.config.ts:27]: Is it intentional to remove this?

File: package.json [hunk 2]
```diff
@@ -180,6 +179,7 @@
   "devDependencies": {
     "@biomejs/biome": "^1.5.3",
     "@pmmmwh/react-refresh-webpack-plugin": "0.5.10",
+    "@sentry/babel-plugin-component-annotate": "^2.13.0",
```
Comment (billyvg) [inline package.json:182]: This will need to be in `dependencies` since we require it to build in prod

File: babel.config.ts (hunk 1 above)
Comment (0Calories) [reply babel.config.ts:27]: Whoops, forgot to put this one back in while I was experimenting. Not intentional

**Review by billyvg — APPROVED** [review]

**Pull request merged**